Tool/Sources/SuggestionProvider/SuggestionProvider.swift

```diff
@@ -47,6 +47,7 @@ public protocol SuggestionServiceProvider {
     func notifyChangeTextDocument(fileURL: URL, content: String) async throws
     func notifyCloseTextDocument(fileURL: URL) async throws
     func notifySaveTextDocument(fileURL: URL) async throws
+    func openChat() async throws
```

Comment:
The reason I added this as a separate command called open chat rather than chat with selection was to distinguish between chatting with ChatGPT vs Codeium Chat. Can you explain further how would this look like if I were to move this out?

Reply:
I would suggestion adding a picker in the chat settings to allow users to select what the open chat command will do. And handle it in `func chatWithSelection(editor: EditorContent)` in WindowBaseCommandHandler.swift.

It's just not recommended to add dynamic items to the Xcode menu because it can't be refreshed until Xcode is restarted.

Reply:
I also found that the openChat method is added to [SuggestionWorkspacePlugin.swift](https://github.com/intitni/CopilotForXcode/pull/485#diff-575fd984ae677336346614e6ec1b715f6321f67bc19f2204ab7f25ce59456bf7), Please create a `CodeiumWorkspacePlugin.swift` and register it in Service.swift

Reply:
Actually there is no API suitable for accessing the CodeiumService in another workspace plugin. Let's keep the changes in SuggestionProvider/WorkspacePlugin in this pr and only change the command.

Reply:
@intitni just to be clear, you are suggesting continuing to use the suggestion service but only change the command to use ChatWithSelection with a toggle that allows users to either use github copilot or codeium as a chat method correct?

Reply:
Correct.

> either use github copilot or codeium as a chat method

Please use words like builtin instead of GitHub Copilot because we don't support GitHub Copilot chat at all.